### PR TITLE
Add stricter few shot validation

### PR DESCRIPTION
**Pull request by petersalas**
Add validation that:
- The func/agent referenced by `ASK_*` lines match those of the ensuring `*_SAYS` lines
- Funcs referenced by lines are registered on the agent
- Non-conversational few-shots have exactly one Q/A
- Conversational few-shots have no patterns between an A and an ensuring Q
- New embeds may not be introduced in `Ask *` or `A:` lines

### CODE REVIEW

File: fixieai/agents/utils.py [hunk 1]
```diff
@@ -125,42 +132,46 @@ def _validate_base_prompt(base_prompt: str):
 
 class FewshotLinePattern(enum.Enum):
     QUERY = re.compile(r"^Q:")
-    AGENT_SAYS = re.compile(r"^Agent\[\w+] says:")
-    FUNC_SAYS = re.compile(r"^Func\[\w+] says:")
-    ASK_AGENT = re.compile(r"^Ask Agent\[\w+]:")
-    ASK_FUNC = re.compile(r"^Ask Func\[\w+]:")
+    AGENT_SAYS = re.compile(r"^Agent\[(\w+)] says:")
```
Comment (hessamb) [inline fixieai/agents/utils.py:135]: optional nit: consider giving these groups a name `(?P<agent_id>\w+)` to use below `match.group('agent_id')`

**Review by hessamb — COMMENTED** [review]
Nice work!

File: fixieai/agents/utils.py [hunk 2]
```diff
@@ -125,42 +132,46 @@ def _validate_base_prompt(base_prompt: str):
...
-        elif len(pattern_matches) == 0:
-            return cls.NO_PATTERN
+        elif len(matches) == 0:
+            return None
 
-        pattern = pattern_matches[0]
-        if pattern is cls.QUERY:
-            match = pattern.value.match(line)
+        match = matches[0]
+        if match.re is cls.QUERY:
```
[the remark this answers is not in the document]
Comment (petersalas) [reply fixieai/agents/utils.py]: Great eye! Yes.

File: fixieai/agents/utils.py (hunk 1 above)
Comment (petersalas) [reply fixieai/agents/utils.py]: That's nicer, thank you!

**Review by hessamb — APPROVED** [review]

**Pull request merged**